apis/projectcontour/v1alpha1/contourconfig_helpers.go

```diff
@@ -185,6 +185,27 @@ func ValidateTLSProtocolVersions(min, max string) error {
 	return nil
 }
 
+// isValidTLSCipher parses a cipher string and returns true if it is valid.
+// We do not support the full syntax defined in the BoringSSL documentation,
+// see https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#Cipher-suite-configuration
+func isValidTLSCipher(cipherSpec string) bool {
+	// Equal-preference group: [cipher1|cipher2|...]
+	if strings.HasPrefix(cipherSpec, "[") && strings.HasSuffix(cipherSpec, "]") {
+		for _, cipher := range strings.Split(cipherSpec[1:len(cipherSpec)-1], "|") {
```

Comment:
nit: could use `strings.Trim` instead of the slice indexing here just to be safe: https://pkg.go.dev/strings#Trim